pkg/security/secl/model/process_cache_entry.go

```diff
@@ -39,11 +39,31 @@ func (pc *ProcessCacheEntry) SetAncestor(parent *ProcessCacheEntry) {
 	parent.Retain()
 }
 
+// GetNextAncestorBinary returns the first ancestor with a different binary
+func (pc *ProcessContext) GetNextAncestorBinary() *ProcessContext {
```

Comment:
I think can have a single implementation of this as ProcessCacheEntry contains a ProcessContext